### PR TITLE
added own LazyLoadingGhostFactory to make directory configureable in DI

### PR DESCRIPTION
**Pull request by danrot**

### CODE REVIEW

File: lib/Subscriber/Core/RegistratorSubscriber.php [hunk 1]
```diff
@@ -218,6 +218,7 @@ public function handleClear(ClearEvent $event)
      *
      * @param AbstractMappingEvent $event
      */
+
```
Comment (wachterjohannes) [inline lib/Subscriber/Core/RegistratorSubscriber.php]: remove this empty line

File: lib/LazyLoadingGhostFactory.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ */
+class LazyLoadingGhostFactory extends BaseLazyLoadingGhostFactory
+{
+    public function __construct($cacheDirectory)
+    {
+        if (!is_dir($cacheDirectory)) {
+            mkdir($cacheDirectory);
+        }
+
+        $configuration = new Configuration();
```
Comment (wachterjohannes) [inline lib/LazyLoadingGhostFactory.php:27]: couldn`t we build this objectwith DI?

File: lib/LazyLoadingGhostFactory.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * Extends the LazyLoadingGhostFactory of the ProxyManager, to make it configureable according to the needs of Sulu.
+ */
+class LazyLoadingGhostFactory extends BaseLazyLoadingGhostFactory
+{
+    public function __construct($cacheDirectory)
+    {
+        if (!is_dir($cacheDirectory)) {
+            mkdir($cacheDirectory);
```
Comment (dantleech) [inline lib/LazyLoadingGhostFactory.php]: Note that there was a recent security vulnerability in Doctrine whereby cache files created with the permissions 777 could be modified by any user on the system to execute any code as the web user. We should probably review Sulu for this security problem generally.

File: lib/Subscriber/Core/RegistratorSubscriber.php (hunk 1 above)
Comment (danrot) [inline lib/Subscriber/Core/RegistratorSubscriber.php]: This line was introduced by styleci... Have to check why this happens...

File: lib/LazyLoadingGhostFactory.php (hunk 3 above)
Comment (danrot) [inline lib/LazyLoadingGhostFactory.php]: I have checked it, the permissions are fine: 
image

File: lib/LazyLoadingGhostFactory.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * Extends the LazyLoadingGhostFactory of the ProxyManager, to make it configureable according to the needs of Sulu.
+ */
+class LazyLoadingGhostFactory extends BaseLazyLoadingGhostFactory
+{
+    public function __construct($cacheDirectory)
+    {
+        if (!is_dir($cacheDirectory)) {
+            mkdir($cacheDirectory, 0777, true);
```
Comment (dantleech) [inline lib/LazyLoadingGhostFactory.php:24]: `777` is the issue here: http://www.doctrine-project.org/2015/08/31/security_misconfiguration_vulnerability_in_various_doctrine_projects.html but it should probably be fixed globally in another PR.

Also, why not remove this class and just configure the `Configuration` in the DI container and set the cache directory there? The cache directory would typically already exist (e.g. `app/cache`).

**Comment by danrot** [comment]
Obsolete because of change suggested in https://github.com/sulu-io/sulu/pull/1576#discussion_r39388514
